geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/ConfigurePDXCommand.java

```diff
@@ -96,59 +72,56 @@ public Result configurePDX(
+      ird.addLine(CliStrings.CONFIGURE_PDX__PERSISTENT + " = false");
     }
 
-    // Set read-serialized
-    if (readSerialized != null) {
-      cache.setPdxReadSerialized(readSerialized);
-    } else {
-      cache.setPdxReadSerialized(CacheConfig.DEFAULT_PDX_READ_SERIALIZED);
-    }
-
-    ird.addLine(CliStrings.CONFIGURE_PDX__READ__SERIALIZED + " = " + cache.getPdxReadSerialized());
-
-    // Set ignoreUnreadFields
-    if (ignoreUnreadFields != null) {
-      cache.setPdxIgnoreUnreadFields(ignoreUnreadFields);
-    } else {
-      cache.setPdxIgnoreUnreadFields(CacheConfig.DEFAULT_PDX_IGNORE_UNREAD_FIELDS);
-    }
-
-    ird.addLine(
-        CliStrings.CONFIGURE_PDX__IGNORE__UNREAD_FIELDS + " = " + cache.getPdxIgnoreUnreadFields());
+    ird.addLine(CliStrings.CONFIGURE_PDX__READ__SERIALIZED + " = " + readSerialized);
+    ird.addLine(CliStrings.CONFIGURE_PDX__IGNORE__UNREAD_FIELDS + " = " + ignoreUnreadFields);
 
     // Auto Serializer Configuration
     if (portableClassesPatterns != null) {
       autoSerializer = createReflectionBasedAutoSerializer(true, portableClassesPatterns);
-      cache.setPdxSerializer(autoSerializer);
-      ird.addLine("PDX Serializer = " + cache.getPdxSerializer().getClass().getName());
+      ird.addLine("PDX Serializer = " + autoSerializer.getClass().getName());
       ird.addLine("Portable Classes = " + Arrays.toString(portableClassesPatterns));
     }
 
     if (nonPortableClassesPatterns != null) {
       autoSerializer = createReflectionBasedAutoSerializer(false, nonPortableClassesPatterns);
-      cache.setPdxSerializer(autoSerializer);
-      ird.addLine("PDX Serializer = " + cache.getPdxSerializer().getClass().getName());
+      ird.addLine("PDX Serializer = " + autoSerializer.getClass().getName());
       ird.addLine("Non Portable Classes = " + Arrays.toString(nonPortableClassesPatterns));
     }
 
-    XmlEntity xmlEntity = createXmlEntity(cache);
     result = ResultBuilder.buildResult(ird);
-    persistClusterConfiguration(result,
-        () -> ((InternalClusterConfigurationService) getConfigurationService())
-            .addXmlEntity(xmlEntity, null));
+    ReflectionBasedAutoSerializer finalAutoSerializer = autoSerializer;
+    getConfigurationService().updateCacheConfig("cluster", config -> {
+      if (config.getPdx() == null) {
+        config.setPdx(new PdxType());
+      }
+      config.getPdx().setReadSerialized(readSerialized);
+      config.getPdx().setIgnoreUnreadFields(ignoreUnreadFields);
+      config.getPdx().setDiskStoreName(diskStore);
+      config.getPdx().setPersistent(diskStore != null ? true : false);
+
+      if (portableClassesPatterns != null || nonPortableClassesPatterns != null) {
+        PdxType.PdxSerializer pdxSerializer = new PdxType.PdxSerializer();
+        pdxSerializer.setClassName(ReflectionBasedAutoSerializer.class.getName());
+        if (finalAutoSerializer != null) {
```

Comment:
This is always true.  `finalAutoSerializer` is initialized if and only if `portableClassPatterns != null`.

Reply:
Done